src/goto-harness/recursive_initialization.cpp

```diff
@@ -38,20 +40,27 @@ void recursive_initializationt::initialize(
   }
   else if(type.id() == ID_pointer)
   {
-    if(
-      lhs.id() == ID_symbol &&
-      should_be_treated_as_array(to_symbol_expr(lhs).get_identifier()))
+    if(lhs.id() == ID_symbol)
     {
-      initialize_dynamic_array(lhs, depth, known_tags, body);
-    }
-    else
-    {
-      initialize_pointer(lhs, depth, known_tags, body);
+      auto const &lhs_symbol = to_symbol_expr(lhs);
+      if(should_be_treated_as_cstring(lhs_symbol.get_identifier()))
+      {
+        initialize_cstring(lhs, depth, known_tags, body);
```

Comment:
Why not call `initialize_dynamic_array` with `cstring_member_initialization` directly here?

Reply:
I think it's better documentation of intent than just calling `initialize_dynamic_array`.

Reply:
Ok, I guess I only wanted to know if it was deliberate.